Try: Update template titles

jameskoster opened the pull request:
Part of https://github.com/WordPress/gutenberg/issues/50395

## What?
Update template titles. 

## Why?
See https://github.com/WordPress/gutenberg/issues/47551, https://github.com/WordPress/gutenberg/issues/44524. In short: Template names are not always particularly intuitive. This PR sacrifices a little brevity in order to (potentially) address that.

## How?
Update `title`s in `$default_template_types`. Here's the full list of changes:

* Home → Blog Page
* Front Page → Homepage
* Singular → Single Entries
* Page → Pages
* Archive → All Archives
* Category → Category Archives
* Tag → Tag Archives
* Date → Date Archives
* Author → Author Archives
* Media → Attachment Pages
* Search → Search Results
* 404 → Page: 404

## Testing Instructions
* Open any UI where template names are encountered, e.g.: Templates List, Add template modal, Template Inspector in the Editor
* Observe that core templates have new names

## Screenshots
Note the names in the modal, and in the sidebar.

### Before
<img width="1364" alt="Screenshot 2023-06-12 at 16 53 11" src="https://github.com/WordPress/gutenberg/assets/846565/6cbaad3b-e06d-4b84-ac1c-9355d85904ed">


### After
<img width="1362" alt="Screenshot 2023-06-12 at 16 52 50" src="https://github.com/WordPress/gutenberg/assets/846565/e9ddd59e-c7e7-4f08-b431-2e210e443c25">


Caveat: Any theme templates that have been customised will retain their old names. I'm not sure how to address that, if we should, or if it's even possible.

jameskoster committed:
Update template titles

alexstine commented on lib/compat/wordpress-6.2/block-template-utils.php:
```diff
@@ -137,7 +137,7 @@ function gutenberg_get_default_block_template_types( $default_template_types ) {
 	if ( isset( $default_template_types['singular'] ) ) {
 		$default_template_types['singular'] = array(
-			'title'       => _x( 'Singular', 'Template name', 'gutenberg' ),
+			'title'       => _x( 'Single Entires', 'Template name', 'gutenberg' ),
```
```suggestion
			'title'       => _x( 'Single Entries', 'Template name', 'gutenberg' ),
```

alexstine commented on lib/compat/wordpress-6.2/block-template-utils.php:
```diff
@@ -146,19 +146,19 @@ function gutenberg_get_default_block_template_types( $default_template_types ) {
 	if ( isset( $default_template_types['single'] ) ) {
 		$default_template_types['single'] = array(
-			'title'       => _x( 'Single', 'Template name', 'gutenberg' ),
+			'title'       => _x( 'Single Posts', 'Template name', 'gutenberg' ),
```
```suggestion
			'title'       => _x( 'Single Post', 'Template name', 'gutenberg' ),
```

jameskoster replied:
For this exercise I considered the template names in terms of what they 'apply to', hence using the plural form here. The singular form makes it _slightly_ less obvious that the template will apply to _all_ posts. 

This format is also useful in the add-template flow, if we frame that as: "What would you like to apply the template to?", there would be options:

* Single posts
* Choose a post

You can see those designs in https://github.com/WordPress/gutenberg/issues/50395.

alexstine commented on lib/compat/wordpress-6.2/block-template-utils.php:
```diff
@@ -146,19 +146,19 @@ function gutenberg_get_default_block_template_types( $default_template_types ) {
 	if ( isset( $default_template_types['page'] ) ) {
 		$default_template_types['page'] = array(
-			'title'       => _x( 'Page', 'Template name', 'gutenberg' ),
+			'title'       => _x( 'Pages', 'Template name', 'gutenberg' ),
```
```suggestion
			'title'       => _x( 'Single Page', 'Template name', 'gutenberg' ),
```

alexstine commented on lib/compat/wordpress-6.2/block-template-utils.php:
```diff
@@ -167,7 +167,7 @@ function gutenberg_get_default_block_template_types( $default_template_types ) {
 	if ( isset( $default_template_types['author'] ) ) {
 		$default_template_types['author'] = array(
-			'title'       => _x( 'Author', 'Template name', 'gutenberg' ),
+			'title'       => _x( 'Author Archives', 'Template name', 'gutenberg' ),
```
```suggestion
			'title'       => _x( 'Single Author Post Archive', 'Template name', 'gutenberg' ),
```

jameskoster replied:
Similar to the comment about plurals above, I think the use of 'Single' here can potentially be a bit confusing.

I don't know that we need to include 'post'... it feels like a technicality that can be defined in the description.

alexstine commented on lib/compat/wordpress-6.2/block-template-utils.php:
```diff
@@ -209,13 +209,13 @@ function gutenberg_get_default_block_template_types( $default_template_types ) {
 	if ( isset( $default_template_types['attachment'] ) ) {
 		$default_template_types['attachment'] = array(
-			'title'       => _x( 'Media', 'Template name', 'gutenberg' ),
+			'title'       => _x( 'Attachment Pages', 'Template name', 'gutenberg' ),
```
```suggestion
			'title'       => _x( 'Single Attachment Page', 'Template name', 'gutenberg' ),
```

alexstine reviewed (commented):
Couple suggestions. Might not want to implement all of them. Naming is hard.

jameskoster committed:
Fix typo

Co-authored-by: Alex Stine <alex.stine@yourtechadvisors.com>

jameskoster commented:
Thanks @alexstine!

jasmussen commented:
Thanks for taking a stab at this. To Alex' point, yes, naming is hard. I'm in general on board with most of these, but the three that stand out to me — and this is almost certainly due to my historic knowledge of WordPress — is Front Page, and Index. I'm not convinced "Homepage" is better than "Frontpage". "All Posts & Pages" instead of "Index" is unclear to me, but that may betray a technical gap on my end: shouldn't it just be "All Posts"?

jameskoster commented:
> I'm not convinced "Homepage" is better than "Frontpage". 

I think the main argument for "homepage" is that being the term used in reading settings, and the front-page template will always display the homepage regardless of settings. "Homepage" feels like it might be a more familiar term in general, but that may be language dependent?

Index is probably the trickiest to get right. There are two angles to capture:

1. Index will display the Posts/Blog page when `home` doesn't exist.
2. It can also resolve to display _anything_ else – posts, pages, archives, even 404.

"All Posts" covers the first angle, but not the second. Perhaps something like "Base" or "Default" 🤔 

Or, we could just leave it alone. The technical name may not be so bad considering:

1. End users will never need to add this template manually, it's required for all themes.
2. Assuming the active theme includes: archive, single-post, page, home, 404, search (fairly common) folks wouldn't ever need to edit it either. 

We might even de-emphasise / hide Index when the conditions are right, but that's one to explore separately.

jasmussen commented:
I'd be tempted to leave it alone, at least for now, but will also acknowledge that this is my template hierarchy and historical "template names" lore that gives me bias.

> "All Posts" covers the first angle, but not the second.

The second use case seems exotic to me, and something we could potentially address in the template description, rather than the title itself?

jameskoster committed:
Revert Index rename

jameskoster commented:
I reverted the change to Index.

jameskoster committed:
"front page" -> "homepage"

paaljoachim commented:
It would also be nice to leave Front page as it is.

Going through the list.

Home → Blog Page 
My comment: That does sound easier to understand.

Front Page → Homepage 
My comment: I would just leave it as Front Page. Front page is pretty clear as it is.

Singular → Single Entries 
My comment: This would be post or a CPT template. Single Entries seems alright. As it has to do with multiple entries.

Page → Pages 
My comment: Single or multiple... as the Page template has to do with all pages than Pages might work fine.

Archive → All Archives
My comment: This seems fine.

Category → Category Archives
My comment: This seems fine.

Date → Date Archives
My comment: This seems fine.

Media → Attachment Pages
My comment: Not sure about this one. I am hesitant. 

Search → Search Results
My comment: This seems fine.

404 → Error: 404
My comment: This seems fine.

I believe there are a few people whom might want to know about this PR.
@carolinan @ndiego

hanneslsm commented:
With naming I like to do the exersize where you start a sentence and you have to end it - it helps with the consistency. This would also help with the discussion on singular or plural.
We only have to agree on the sentence. I might make sense, feel also free to ignore it, though. 
Here is a suggestion:
**This template is for…**
**This template displays…**
- Blog Pages
- (the) Front Page
- Posts *?*
- Pages
- Post Archives *?*
- Category Archives *?*
- Date Archives *?*
- Media Pages *?*
- Search Results
- (the) Error Page (404) *?*

jameskoster committed:
Update 404

Page make more sense given the template appears in the Pages panel too.

jameskoster commented:
Including 'page' in the 404 title could work given it appears in the Pages panel too.

jameskoster commented:
On a similar train of thought, I wonder if it might be worth considering leaning more into 'page' as a term rather than 'archive', where appropriate. 'Page' is probably a bit friendlier, and the template hierarchy does refer to many templates as pages, so there's precedent. 

Instead of:

* Category Archives
* Tag Archives
* Author Archives
* Archives
* Search results

We might try:

* Category Pages
* Tag Pages
* Author Pages
* Archive Pages
* Search Results Page

alexstine commented:
Just be careful conflating pages. As in page.php or frontpage.php are also page templates.

paaljoachim commented:
That would create too much focus on the word page. It is better to keep the word archive. As in a sense that is what it is.

carolinan commented:
I think that
```
Home → Blog Page
Front Page → Homepage
```
Will make it easier for any new user who has never needed to figure out what the "home" template is.
But for anyone who is already using WordPress and has made edits to home.php or home.html, I think this is more confusing.

I think it will be confusing if the HTML file names do not match the title. How will beginner developers who want to 
find the file for the **blog page** know that it is called home.html, and the Homepage is called front-page.html?

jameskoster commented:
@carolinan yes that's the tricky part to balance. It's a question about who the UI should be optimised for, technical users (theme developers), or non-technical users. The current setup caters more to the former, but in doing so makes things a bit more confusing than they might be for the majority. 

I didn't mock it up yet, but it may be reasonable to include the technical name somewhere in the template details panel as a hint for developers?

jameskoster commented:
> I think it will be confusing if the HTML file names do not match the title

A couple of follow-up thoughts on this. If the theme is built entirely in the Editor (not saying that's generally what happens right now, but it's the direction we're headed in), then isn't it possible a theme author will never actually encounter the files? 

For an end user there will also be a number of templates that exist only in the database, with no link to the file system at all.

In situations where there _is_ a corresponding file, perhaps we can include that in the template details panel somehow:

<img width="727" alt="Screenshot 2023-06-21 at 11 57 38" src="https://github.com/WordPress/gutenberg/assets/846565/134b4f8c-a397-4039-a351-6e8ba5832dab">

We may consider only displaying this when `WP_DEBUG` is `true`. What do you think?

cc @WordPress/block-themers for any more feedback on this PR.

jasmussen commented:
Taking stock:

* The renamed files will definitely benefit casual non-developery users with all the more clarity
* The developer angle that the names may go out of sync with what the files are called according to their historical names is also valid, especially in the two examples mentioned, such as the Home and Front Page renames.

Can we do the following?

* In the template creation modal, show the historical file as a tooltip, such as hovering Blog page would show `Displays the posts page, also known as "Home"`
* In the detail page, instead of showing the filename/historical name in the ellipsis menu, work it into the description itself. It could be something like `Displays the Posts page, also known as "Home" ...` plus the rest of the description. 

The description becomes a bit more verbose, but I don't think that's a problem, in fact it's part of the point of the detail pages that they can afford this pertinent information.

If we do those changes, I think we can land this rename in time for 6.3. This gives us the beta phase to then listen to feedback, and revert if we need to. What do you think?

MaggieCabrera commented:
> We may consider only displaying this when `WP_DEBUG` is `true`. What do you think?
> 
> cc @WordPress/block-themers for any more feedback on this PR.

That sounds very sensible to me

jameskoster commented:
@jasmussen I think that's a good point about the Home template. It's really the only one where a developer might not make the connection immediately. Things like "Date archives" → `date.html` are fairly intuitive. I'll try to work that into the description.

jameskoster committed:
Front Page

jameskoster committed:
home title + description

jameskoster commented:
I pushed a (hopefully) simpler solution. "Blog page" → "Blog home page". What do you think?

I'm not sure, but a core patch may be required for this. Perhaps @ntsekouras knows?

carolinan commented:
I agree the description works better than the ellipsis menu, because that menu is mostly used for actions, not information.

jameskoster commented:
@carolinan how do you feel about the updated title instead? It's a little tricky to fit the technical name into the current description, but I would welcome any ideas:

> Displays the latest posts as either the site homepage or a custom page defined under reading settings. If it exists, the Front Page template overrides this template when posts are shown on the front page.

richtabor commented:
Should "Homepage" just be "Home" and the "Blog page" be "Blog"? 

"Page" isn't appended to other template titles.

jasmussen commented:
Also important to decide if it's title case or sentence case across.

jameskoster commented:
>"Blog page" be "Blog"?

That's nice and neat. There's a chance that a user might think it applies to _all_ blog pages, but hopefully the description clears that up. Happy to try it. 

> Should "Homepage" just be "Home"?

I'm not sure about literally using a technical template name to represent a different template. In this case "Home" would equal `front-page.html`. I suspect a developer would interpret "Home" = `home.html`.

ntsekouras commented:
>I'm not sure, but a core patch may be required for this. Perhaps @ntsekouras knows?

Since we're only updating titles and descriptions, they will be part of the back porting process. In general the changes should be moved to `lib/compat/wordpress-6.3/block-template-utils.php` and not `wordpress-6.2` folder. If this PR doesn't make it for `6.3`, the changes should be moved to a respective file for `6.4`.

jameskoster committed:
Revert Home name change, and move compat files

jameskoster commented:
The Home template seems to be the most contentious, so I've reverted that one. We can investigate it in a dedicated PR. 

@ntsekouras I've moved the compatibility file to the suggested location. If you could just check that I'd appreciate it.

jasmussen reviewed (approved):
I'm happy to try this. It also benefits that this sorts the 404 page lower.

ntsekouras commented:
>@ntsekouras I've moved the compatibility file to the suggested location. If you could just check that I'd appreciate it.

I'm going to push some small changes regarding this.

ntsekouras committed:
separate code for wp versions

ntsekouras reviewed (approved):
That's ready to land now 😄 . Thank you!

ntsekouras committed:
update tests

tellthemachines commented:
I just cherry-picked this PR to the add/pre-beta1-bugfixes branch to get it included in the next release: 449ec381d2

annezazu commented:
@tellthemachines this was supposed to make it into beta 1 but I am still seeing in beta 3 the following (with 404 listed up top):

<img width="351" alt="Screen Shot 2023-07-10 at 10 09 52 AM" src="https://github.com/WordPress/gutenberg/assets/26996883/c005fd0e-97dd-48c4-a615-0a079104c530">


If I use 6.3-beta3-56182, I'm able to correctly see the following:

<img width="352" alt="Screen Shot 2023-07-10 at 10 08 05 AM" src="https://github.com/WordPress/gutenberg/assets/26996883/3682fd42-e400-40ad-a848-6a3a1f5e13b4">

Can you confirm what's going on here and ensure it makes it appropriately into beta 4?

jameskoster commented:
@annezazu something I've noticed happening intermittently is: if you customised a template like 404 _before_ the name change, it seems to retain that name in perpetuity. But if you Clear customizations it will 'revert' to the new name. Might be worth investigating.

tellthemachines commented:
The core patch was [committed last Thursday](https://github.com/WordPress/wordpress-develop/pull/4793#issuecomment-1622891282) so it's in core trunk now, ready to go into Beta 4.

annezazu commented:
Excellent. Thank you for confirming!
